### PR TITLE
Feature/cli usable as package

### PR DESCRIPTION
**Pull request by sbnetops**
Including a class `Polyswarm` that extends from `PolyswarmAPI` and adds extra methods that provide threaded parallelism.

### CODE REVIEW

File: src/polyswarm/__main__.py [hunk 1]
```diff
@@ -1,5 +1,20 @@
...
-from .base import polyswarm
+from __future__ import absolute_import
+import os
+import sys
+from polyswarm.client.polyswarm import polyswarm
+
+
+def main():
+    # fixing the name of the entrypoint when the module is executed as
+    # python -m s3conf
```
Comment (mrtizmoatwork) [inline src/polyswarm/__main__.py:10]: what is s3conf?

File: src/polyswarm/polyswarm.py [hunk 2]
```diff
@@ -0,0 +1,258 @@
...
+        """
+        args = [(u,) for u in url]
+        for instance in utils.parallel_executor_iterable_results(self.search_url, args_list=args):
+            yield instance
+
+    def download_multiple(self, values, destination, hash_type=None):
+        """
+        Download files from matching hashes.
+
+        :param values: A list of hashes to download.
```
Comment (mrtizmoatwork) [inline src/polyswarm/polyswarm.py:161]: can we name "values" as "hashes", it would make a lot more sense.

File: src/polyswarm/polyswarm.py [hunk 3]
```diff
@@ -0,0 +1,258 @@
...
+        kwargs = [dict(timeout=timeout)]*len(args)
+
+        for result in utils.parallel_executor(self.wait_for, args_list=args, kwargs_list=kwargs):
+            yield result
+
+    def search_hashes(self, values, hash_type=None):
+        """
+        Search PolySwarm for files matching hashes.
+
+        :param values: List of hashes to search.
```
Comment (mrtizmoatwork) [inline src/polyswarm/polyswarm.py:137]: rename "values" as "hashes"

File: src/polyswarm/__main__.py (hunk 1 above)
Comment (sbnetops) [reply src/polyswarm/__main__.py:10]: Ops, sorry, this other codebase of mine which I used as template.

Comment (alanjds) [inline src/polyswarm/__main__.py:10]:
```suggestion
    # python -m s3conf
    # python -m polyswarm
```

File: src/polyswarm/client/download.py [hunk 4]
```diff
@@ -24,12 +25,10 @@ def download(ctx, hash_file, hash_type, hash_value, destination):
...
     """
     api = ctx.obj['api']
-    output = ctx.obj['output']
-    args = [(destination, h) for h in utils.parse_hashes(hash_value, hash_file=hash_file)]
-
-    for result in utils.parallel_executor(api.download, args_list=args,
-                                          kwargs_list=[{'hash_type': hash_type}]*len(args)):
-        output.local_artifact(result)
+    out = ctx.obj['output']
+    values = utils.parse_hashes(hash_value, hash_file=hash_file)
```
Comment (mrtizmoatwork) [inline src/polyswarm/client/download.py:29]: "values" -> "hashes"

File: src/polyswarm/client/scan.py [hunk 5]
```diff
@@ -105,13 +78,8 @@ def rescan(ctx, hash_file, hash_type, timeout, nowait, hash_value, scan_config):
...
     api = ctx.obj['api']
     output = ctx.obj['output']
-
-    args = [(api, timeout, nowait, h) for h in utils.parse_hashes(hash_value, hash_file=hash_file)]
-
-    for instance in utils.parallel_executor(rescan_and_wait,
-                                            args_list=args,
-                                            kwargs_list=[{'hash_type': hash_type,
-                                                          'scan_config': scan_config}]*len(args)):
+    values = utils.parse_hashes(hash_value, hash_file=hash_file)
```
Comment (mrtizmoatwork) [inline src/polyswarm/client/scan.py:81]: "values" -> "hashes"

File: src/polyswarm/client/search.py [hunk 6]
```diff
@@ -30,10 +31,8 @@ def hashes(ctx, hash_value, hash_file, hash_type):
     """
     api = ctx.obj['api']
     output = ctx.obj['output']
-
-    args = [(h,) for h in utils.parse_hashes(hash_value, hash_file=hash_file)]
-    for instance in utils.parallel_executor_iterable_results(api.search, args_list=args,
-                                                             kwargs_list=[{'hash_type': hash_type}]*len(args)):
+    values = utils.parse_hashes(hash_value, hash_file=hash_file)
```
Comment (mrtizmoatwork) [inline src/polyswarm/client/search.py:34]: "values" -> "hashes"

File: src/polyswarm/client/utils.py [hunk 7]
```diff
@@ -0,0 +1,81 @@
...
+
+logger = logging.getLogger(__name__)
+HASH_VALIDATORS = resources.Hash.SUPPORTED_HASH_TYPES
+
+####################################################
+# Input parsers
+####################################################
+
+
+def parse_hashes(values, hash_file=None):
```
Comment (mrtizmoatwork) [inline src/polyswarm/client/utils.py:19]: can do "values" -> "hashes" 
and
"hashes" -> "hashes_"

**Pull request merged**